src/parser/warlock/affliction/modules/azerite/SuddenOnset.js

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+
+import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events from 'parser/core/Events';
+import StatTracker from 'parser/shared/modules/StatTracker';
+import calculateBonusAzeriteDamage from 'parser/core/calculateBonusAzeriteDamage';
+
+import SPELLS from 'common/SPELLS';
+import { calculateAzeriteEffects } from 'common/stats';
+import { formatThousands } from 'common/format';
+
+import TraitStatisticBox from 'interface/others/TraitStatisticBox';
+import ItemDamageDone from 'interface/others/ItemDamageDone';
+
+const AGONY_SP_COEFFICIENT = 0.008; // taken from Simcraft SpellDataDump
+
+/*
+    Sudden Onset
+      Agony deals up to an additional X damage and starts with 4 stacks.
+ */
+class SuddenOnset extends Analyzer {
+  static dependencies = {
+    statTracker: StatTracker,
+  };
+
+  traitBonus = 0;
+  damage = 0;
+
+  constructor(...args) {
+    super(...args);
+    this.active = this.selectedCombatant.hasTrait(SPELLS.SUDDEN_ONSET.id);
+    if (!this.active) {
+      return;
+    }
+    this.traitBonus = this.selectedCombatant.traitsBySpellId[SPELLS.SUDDEN_ONSET.id].reduce((total, rank) => {
```

Comment:
is the value of the trait the bonus dmg at max stacks? If that is the case you might need to multiply the sp coef by max stack count (10?).

Reply:
No, I've been told this is the base damage bonus, and the stacking mechanic of Agony is just another multiplier on top of Mastery, Versatility, spec aura etc. So additional stacks will scale the base damage bonus further same as normal Agony damage.